tests/lib/prepare-uc20.sh

```diff
@@ -258,26 +258,46 @@ echo "ubuntu:x:1001:" >> /root/test-var/lib/extrausers/group
 sed -r -i -e 's/^systemd-journal:x:([0-9]+):$/systemd-journal:x:\1:test/' /root/test-etc/group
 
 # mount fresh image and add all our SPREAD_PROJECT data
-kpartx -avs pc.img
-# losetup --list --noheadings returns:
-# /dev/loop1   0 0  1  1 /var/lib/snapd/snaps/ohmygiraffe_3.snap                0     512
-# /dev/loop57  0 0  1  1 /var/lib/snapd/snaps/http_25.snap                      0     512
-# /dev/loop19  0 0  1  1 /var/lib/snapd/snaps/test-snapd-netplan-apply_75.snap  0     512
-devloop=$(losetup --list --noheadings | grep pc.img | awk '{print $1}')
-dev=$(basename "$devloop")
-
-# mount it so we can use it now
-mkdir -p /mnt
-mount "/dev/mapper/${dev}p2" /mnt
-
-# add the data that snapd.spread-tests-run-mode-tweaks.service reads to the 
-# mounted partition
-tar -c -z \
-    -f /mnt/run-mode-overlay-data.tar.gz \
-    /root/test-etc /root/test-var/lib/extrausers
-
-# tear down the mounts
-umount /mnt
-kpartx -d pc.img
+# for the lxd backend this step is a bit different as the device mapper kernel module
+# is not supported by lxd containers. We thus have to do some manual setup of the image
+# partition mount.
+if [ "${SPREAD_BACKEND}" = "lxd-nested" ]; then
+    devloop=$(losetup -f)
```

Comment:
You can use `-f` directly into the `losetup` command 2 lines under. That will save us from a potential race where something takes that loop device.

Reply:
Ah nice, i did not know about this, i'll update it!